[clj55] iP

clj55 opened the pull request:
Changed Duke to Psyduck

clj55 committed:
Level-0

clj55 committed:
add echo

clj55 committed:
add user-psyduck roles

clj55 committed:
add mark as done

clj55 committed:
Edited code to follow coding style

clj55 committed:
Converted Task into a class

clj55 committed:
Added Events and Deadlines

IgoyAI commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,118 @@
+        Scanner scanObj = new Scanner(System.in);
+        List<Task> taskList = new ArrayList<Task>();
+        System.out.println("Hello! I'm Psyduck");
+        System.out.println("PSYDUCK?");
```
Consider to make it into higher level abstraction such as printWelcomeMessage().

IgoyAI commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,118 @@
+        String userInput = scanObj.nextLine();
+        printDashes();
+        userInput = userInput.trim();
+        while (!userInput.equals("bye")) {
```
Consider to make it into higher level abstraction

IgoyAI commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,118 @@
+            } else if (userInput.startsWith("unmark")) {
+                String sNum = userInput.substring(7);
+                unmarkTask(sNum, taskList);
+            } else if (userInput.startsWith("mark")) {
```
No space after opening braces

clj55 replied:
Do you mean I should remove the space between ) and { ?

IgoyAI commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,46 @@
+public class Task {
```
All good

dylankhoo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,46 @@
+public class Task {
+    private String taskName;
+    private char taskType;
+    private boolean done;
```
Consider renaming bool variable to use standard prefixes (e.g. isDone)

dylankhoo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,46 @@
+        this.taskType = taskType;
+    }
+
+    public boolean isDone() {
```
Try to name methods in verb form. Since the method is a getter, you can consider using the convention (e.g. getDone).

dylankhoo commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,118 @@
+                String sNum = userInput.substring(5);
+                markTask(sNum, taskList);
+            } else if (userInput.startsWith("deadline")) {
+                String s = userInput.substring(8);
```
Consider renaming variable "s" to an English word for readability

dylankhoo commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,118 @@
+                System.out.println("quacker not in list");
+            } else {
+                System.out.println("Psyduck is impressed");
+                iNum -= 1; // adjust to start with zero index
```
Consider rewriting comment to be more intuitive

dylankhoo reviewed (commented):
Coding Standard largely followed. 
Suggested a few changes for renaming variables and methods.

Carlton369 commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,118 @@
+
+    }
+
+    private static void printAddTaskStatement(Task task) {
```
Good naming convention for your methods, very intuitive!

Carlton369 commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,118 @@
+                System.out.println();
+            }
+        } catch (NumberFormatException e) {
+            System.out.println("PSYDUCK ANGRY: PSYDUCK WANT NUMBERS");
```
Very flavorful and intuitive error messages that make the error very explicit for the user!

Carlton369 commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,118 @@
+public class Psyduck {
+    private static int count = 0;
+
+    public static void printDashes() {
```
Good use of simple method to avoid repeating this line of code! Glad to see you are diligently following the DRY principle!

Carlton369 commented on src/main/java/Psyduck.java:
```diff
@@ -0,0 +1,118 @@
+        }
+    }
+
+    private static void unmarkTask(String sNum, List<Task> taskList) {
```
Although this method is already quite readable, adding a comment to describe it could help improve readability (nit)

Carlton369 reviewed (commented):
Overall very impressed with the quality of the code, just a small nit but otherwise very good work! Code is very clean and readable.

clj55 committed:
add todo command

previously todos were automatically added when no command specified, now it must have the todo command

clj55 committed:
fix todo task

clj55 committed:
Extracted more methods

so that it SLAPS harder

clj55 committed:
renamed Done attribute and isDone method

clj55 committed:
Add Exception handling and clean code

clj55 committed:
divided files into packages

clj55 committed:
Merge branch 'branch-Level-5'

clj55 committed:
add Delete Task function

clj55 committed:
add auto saving task data functionality

saves data in a txt file: tasks.txt in data package
if text file does not exist, creates the text file

clj55 committed:
Merge branch 'branch-Level-7'

clj55 committed:
Merge branch 'branch-Level-6'

# Conflicts:
#	src/main/java/psyduck/ui/Psyduck.java

clj55 committed:
Fix Loading of marked tasks and Creates file directory for tasks.txt

clj55 committed:
Fix delete function to delete from tasks.txt file too

aditichadha1310 commented on src/main/java/psyduck/ui/Psyduck.java:
```diff
@@ -0,0 +1,261 @@
+    private static final Scanner userInputScanner = new Scanner(System.in);
+    private static List<String> fileLines = new ArrayList<>();
+
+    public static void main(String[] args) throws IOException {
```
Avoid lengthy methods. try to keep each method to about 30 lines of code.

aditichadha1310 commented on src/main/java/psyduck/ui/Psyduck.java:
```diff
@@ -0,0 +1,261 @@
+        Task task = new Task(details.trim());
+        addtoTaskList(task);
+        appendToFile(TASK_LIST_FILEPATH, task.getTaskType() + "/" +
+                +(task.checkDone() ? 1 : 0) + "/" + task.getTaskName() + "\n");
```
Avoid usage of magic strings and numbers

aditichadha1310 reviewed (commented):
overall, good work!

clj55 committed:
separated functions into different classes

clj55 committed:
corrected main-class back to Psyduck

clj55 committed:
Add Find Command and Catch when user puts too many datetimes in Event or Deadline

clj55 committed:
Add Find Command and Check for too many datetimes in event and deadline

Forgot to stage earlier changes
Simplified printTask to use ToString

clj55 committed:
Add JavaDoc for changed files

clj55 committed:
Fix AddDeadline/Event to show correct error message when datetime not given

clj55 committed:
Add function to create data file automatically

Fix Event Function to save to text file
Format all code

clj55 committed:
Added command format to Error messages

clj55 committed:
Merge branch 'branch-level-9'

# Conflicts:
#	src/main/java/psyduck/command/AddDeadlineCommand.java
#	src/main/java/psyduck/command/AddEventCommand.java
#	src/main/java/psyduck/command/Command.java
#	src/main/java/psyduck/command/DeleteCommand.java
#	src/main/java/psyduck/parser/Parser.java
#	src/main/java/psyduck/task/Deadline.java
#	src/main/java/psyduck/task/Event.java
#	src/main/java/psyduck/task/Task.java

clj55 committed:
Remove typo

clj55 committed:
Handle errors when task file changed improperly

clj55 committed:
Catch Error when users put blanks in Task

clj55 committed:
Add more explanations for functions

clj55 committed:
Merge branch 'A-JavaDoc'

# Conflicts:
#	.gitignore
#	src/main/java/psyduck/command/AddCommand.java
#	src/main/java/psyduck/command/Command.java
#	src/main/java/psyduck/command/MarkCommand.java
#	src/main/java/psyduck/command/UnmarkCommand.java
#	src/main/java/psyduck/exception/TaskUndefinedException.java
#	src/main/java/psyduck/storage/FileInterfacer.java
#	src/main/java/psyduck/storage/TaskFileInterfacer.java
#	src/main/java/psyduck/ui/UI.java

clj55 committed:
Add JavaDoc for Find Command

Remove unnecessary exception

clj55 committed:
Create UserGuide

clj55 committed:
Edit Command Summary Table

clj55 committed:
Reorder and add Spaces to user guide

clj55 committed:
Add Links to user guide

clj55 committed:
Remove minor typos

clj55 committed:
Test change effect on PR

clj55 committed:
revert to working copy
